app/controllers/service_providers_controller.rb

```diff
@@ -197,5 +199,27 @@ def add_iaa_warning
     flash.now[:warning] = I18n.t('notices.service_provider_iaa_notice')
   end
 
+  def clear_formatting(service_provider)
+    string_attributes = %w[
+      issuer
+      friendly_name
+      description
+      metadata_url
+      acs_url
+      assertion_consumer_logout_service_url
+      sp_initiated_login_url
+      return_to_sp_url
+      production_issuer
+      failure_to_proof_url
+      push_notification_url
+      app_name
+    ]
+
+    service_provider.attributes.each do |k,v|
+      v.try(:strip!) unless !string_attributes.include?(k)
+    end
+    service_provider
```

Comment:
a few ideas:
1. let's only iterate the string attributes?
2. I think we could just inline `string_attributes`
3. I have personal beef with `unless` so I prefer `if`
4. I'm worried that iterating the attributes hash like this won't write back to the model?

```suggestion
   %w[
      issuer
      friendly_name
      description
      metadata_url
      acs_url
      assertion_consumer_logout_service_url
      sp_initiated_login_url
      return_to_sp_url
      production_issuer
      failure_to_proof_url
      push_notification_url
      app_name
    ].each do |string_attribute|
      service_provider[string_attribute] = service_provider[string_attribute]&.strip
    end
    service_provider
```

Reply:
I also have a personal beef with unless but rubocop forced me into this.

Reply:
!!! let's remove that rule, we removed it in the IDP a while ago

Reply:
I don't think we can iterate over `string_attributes` because all of them will not exist on each service provider depending on the protocol.  And the test checks that this method is run before an update of the record is returned with properly stripped values.  Further, just using `strip` is will not save to the model and dropping the `try` will blow up on nils which certainly exist in this array of keys depending on the protocol.

Reply:
AFAIK, the attributes are on each SP depending on the protocol but somtimes they're nil

And the `&.` is the new ruby shorthand for `.try` so I think it handles?